pkg/kando/process_client_output.go

```diff
@@ -30,22 +41,74 @@ func newProcessClientOutputCommand() *cobra.Command {
 		Args:  cobra.ExactArgs(1),
 		RunE:  runProcessClientOutput,
 	}
+	cmd.PersistentFlags().BoolP(processSignalProxyFlagName, "p", false, "pass signals from client to server")
 	return cmd
 }
 
+func processSignalProxyFlagValue(cmd *cobra.Command) (bool, error) {
+	return cmd.Flags().GetBool(processSignalProxyFlagName)
+}
+
 func runProcessClientOutput(cmd *cobra.Command, args []string) error {
-	return runProcessClientOutputWithOutput(cmd.OutOrStdout(), cmd, args)
+	return runProcessClientOutputWithOutput(cmd.OutOrStdout(), cmd.ErrOrStderr(), cmd, args)
+}
+
+func proxySetup(ctx context.Context, addr string, pid int64) {
+	log.Info().WithContext(ctx).Print(fmt.Sprintf("signal proxy is running for process %d", pid))
+	signalTermChan := make(chan os.Signal, 1)
+	signal.Notify(signalTermChan, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+	BREAK:
+		for {
+			select {
+			case sig := <-signalTermChan:
+				ossig, ok := sig.(syscall.Signal)
+				if !ok {
+					log.Info().WithContext(ctx).Print(fmt.Sprintf("signal %v is invalid, ignored for process %d", sig, pid))
+					continue
+				}
+				log.Info().WithContext(ctx).Print(fmt.Sprintf("signal %v received for process %d", sig, pid))
+				_, err := kanx.SignalProcess(ctx, addr, pid, int64(ossig))
```

Comment:
one issue here... can kanx reuse a PID number ?

I think linux can, so I'd assume that kanx can too.

Is it possible that you might end up killing the wrong process due to pid reuse ? I'll result in random disconnects in the field for a system that is heavily loaded and is reusing PID numbers.

What do you think ?

Reply:
PID reuse wil be solved later, we discussed this over slack